plugins/modules/vmware_guest.py

```diff
@@ -1935,6 +1921,55 @@ def sanitize_network_params(self):
 
         return network_devices
 
+    def get_network(self, network_name=None, network_device=None):
+        """
+        Get network depending upon parameters provided by user
+
+        """
+        network = None
+        esxi_host = None
+        pg_lookup = {}
+
+        if self.params.get('esxi_hostname'):
+            esxi_host = self.select_host()
+            if esxi_host:
+                for pg in esxi_host.config.network.portgroup:
+                    pg_lookup[pg.spec.name] = {'switch': pg.spec.vswitchName, 'vlan_id': pg.spec.vlanId}
+        dvswitch_name = network_device.get('dvswitch_name', None)
+        vlan_id = network_device.get('vlan', None)
+
+        all_networks = self.cache.get_all_objs(self.content, [vim.Network])
+        for network in all_networks:
+            if esxi_host and isinstance(network, vim.Network):
+                # User specified ESXi and network without dvswitch details
+                # We assume user needs standard portgroup
+                if not dvswitch_name and network_name and network.name == network_name:
+                    # Standard portgroup by name
+                    return network
+                if vlan_id:
+                    # Standard portgroup by vlan id
+                    for k in pg_lookup.keys():
+                        if vlan_id == pg_lookup[k]['vlan_id']:
+                            if k == network.name:
+                                return network
+            elif dvswitch_name and isinstance(network, vim.dvs.DistributedVirtualPortgroup):
+                # Distributed Virtual Portgroup by DV Switch
+                dvs = network.config.distributedVirtualSwitch
+                if dvs.config.name == dvswitch_name and network.config.name == network_name:
```

Comment:
Can you make the network search so that the vlan can be narrowed down by dvswitch? I have a use case where there is more than one dvswitch with the same vlan. Maybe something like:
```
            elif dvswitch_name and isinstance(network, vim.dvs.DistributedVirtualPortgroup):
                # Distributed Virtual Portgroup by DV Switch
                dvs = network.config.distributedVirtualSwitch
                if dvs.config.name == dvswitch_name and (network.config.name == network_name or network.config.defaultPortConfig.vlan.vlanId == vlan_id):
                    return network
```

Looks like in 2.9 this functionality previously existed:
https://github.com/ansible/ansible/blob/stable-2.9/lib/ansible/modules/cloud/vmware/vmware_guest.py#L1367